app/src/sharedTest/java/org/oppia/android/app/spotlight/SpotlightFragmentTest.kt

```diff
@@ -0,0 +1,350 @@
+
+/** Tests for [SpotlightFragment]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = SpotlightFragmentTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class SpotlightFragmentTest {
+  @Inject
+  lateinit var context: Context
+
+//  @get:Rule
+//  val oppiaTestRule = OppiaTestRule()
+
+  @field:[Rule JvmField]
+  val mockitoRule: MockitoRule = MockitoJUnit.rule()
+
+  @Inject
+  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+
+  @Inject
+  lateinit var appLanguageLocaleHandler: AppLanguageLocaleHandler
+
+  @Inject
+  @field:DefaultResourceBucketName
+  lateinit var resourceBucketName: String
+
+//  @get:Rule
+//  var activityScenarioRule: ActivityScenarioRule<SpotlightFragmentTestActivity> =
+//    ActivityScenarioRule(
+//      Intent(
+//        ApplicationProvider.getApplicationContext(),
+//        SpotlightFragmentTestActivity::class.java
+//      )
+//    )
+
+  private val sampleSpotlightText = "Sample spotlight hint text"
+  private val sampleSecondSpotlightText = "Sample hint text for second spotlight"
+
+  @Before
+  fun setUp() {
+    Intents.init()
+    setUpTestApplicationComponent()
+    testCoroutineDispatchers.registerIdlingResource()
+  }
+
+  @After
+  fun tearDown() {
+    Intents.release()
+    testCoroutineDispatchers.unregisterIdlingResource()
+  }
+
+  @Test
+  fun testSpotlightFragment_requestSpotlight_shouldShowSpotlight() {
+    launch<SpotlightFragmentTestActivity>(
+      createSpotlightFragmentTestActivity(context)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      it.onActivity { activity ->
+        val spotlightTarget = SpotlightTarget(
+          activity.getSampleSpotlightTarget(),
+          sampleSpotlightText,
+          SpotlightShape.RoundedRectangle,
+          Spotlight.FeatureCase.PROMOTED_STORIES
+        )
+
+        checkNotNull(activity.getSpotlightFragment()).requestSpotlight(spotlightTarget)
+        testCoroutineDispatchers.runCurrent()
+      }
+      onView(withText(sampleSpotlightText)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testSpotlightFragment_requestDelayedSpotlight_shouldShowSpotlight() {
+    launch<SpotlightFragmentTestActivity>(createSpotlightFragmentTestActivity(context)).use {
+      testCoroutineDispatchers.runCurrent()
+      it.onActivity { activity ->
+        val spotlightTarget = SpotlightTarget(
+          activity.getSampleSpotlightTarget(),
+          sampleSpotlightText,
+          SpotlightShape.RoundedRectangle,
+          Spotlight.FeatureCase.PROMOTED_STORIES
+        )
+
+        checkNotNull(
+          activity.getSpotlightFragment()
+        ).requestSpotlightViewWithDelayedLayout(spotlightTarget)
+        testCoroutineDispatchers.runCurrent()
+      }
+      onView(withText(sampleSpotlightText)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testSpotlightFragment_markSpotlightSeen_checkSpotlightIsNotShowAgain() {
```

Comment:
Suggest also adding a test for verifying that showing the same spotlight twice back-to-back DOES show it again if the first one wasn't closed.

Reply:
done